src/main/java/io/reactiverse/pgclient/impl/codec/DataTypeCodec.java

```diff
@@ -472,16 +463,29 @@ public static Object prepare(DataType type, Object value) {
         } else {
           return REFUSED_SENTINEL;
         }
+      case UNKNOWN:
+        if (value instanceof String[]) {
+          return Arrays.stream((String[]) value).collect(Collectors.joining(",", "{", "}"));
+        } else if (value instanceof String) {
+          return value;
+        } else {
+          return REFUSED_SENTINEL;
+        }
       default:
         Class<?> javaType = type.type;
         return javaType.isInstance(value) ? javaType.cast(value) : REFUSED_SENTINEL;
     }
   }
 
   private static Object defaultDecodeText(int len, ByteBuf buff) {
-    // Default to null
-    buff.skipBytes(len);
-    return null;
+    // decode unknown text values as text or as an array if it begins with `{`
+    if (buff.getByte(buff.readerIndex()) == '{') {
```

Comment:
can this fail because of an empty text value ?

Reply:
Hmm, not entirely sure but I'll err on the side of caution here and add a check

Reply:
there should be a check that the readable bytes size is > 0

Reply:
👍